Updates to install (or upgrade) and run Count.ly Server on macOS

GlennAustin opened the pull request:
I'm still in the process of installing 20.04 on my server, but these are the changes I needed to make to upgrade my 18.08 installation to 19.08.1.

GlennAustin commented:
This has now been tested with a clean install on macOS, and an upgrade from 19.08.1 to 20.04.

GlennAustin commented on bin/upgrade/18.08/upgrade_fs.sh:
```diff
@@ -35,7 +35,7 @@ rm -rf "$DIR/../node_modules"
 
 
-countly upgrade
+sudo countly upgrade
```
Done.

ar2rsawseen replied:
I guess this should be done for all upgrade scripts

GlennAustin replied:
I think I've removed all usages of sudo outside of launchd scripts. Please let me know if I missed any others

GlennAustin commented on bin/upgrade/20.04/upgrade_fs.sh:
```diff
@@ -44,24 +79,29 @@ then
+    bash countly plugin upgrade attribution
+    bash countly plugin upgrade web
+    bash countly plugin enable active_users
+    bash countly plugin enable performance-monitoring
```
Removed

GlennAustin commented:
Requested changes made.

ar2rsawseen commented on bin/upgrade/20.04/upgrade.sh:
```diff
@@ -1,13 +1,55 @@
+        fi
+        OLD_COUNTLY_DIR="$( cd "$OLD_COUNTLY_DIR"/.. && pwd )"
+    done
+fi
```
Yeah, let's not do changes likes this and keep this PR exclusively to needed changes for Mac OSX if possible :)

GlennAustin commented:
- Removed upgrade helper to transfer app icons from old version to new version.
- Removed remaining usages of sudo in general scripts
- Removed pure text-formatting changes (indentations, blank line adds/removes)

GlennAustin commented on bin/commands/countly.sh:
```diff
@@ -10,7 +10,20 @@ done
-VERSION="$(grep -oP 'version:\s*"\K[0-9\.]*' "$DIR/../../frontend/express/version.info.js")"
+VERSION="$(egrep -o 'version: *"[0-9\.]+"' "$DIR/../../frontend/express/version.info.js" | sed -E -e 's,^version: *"([0-9\.]+)"$,\1,')"
+
+#Activate our nodejs, if using nodeenv
```
I'm using nodeenv on macOS since it provides a nice, clean environment for Count.ly without affecting any other NodeJS applications that may be running. I suggest switching over to nodeenv since Count.ly isn't using the latest NodeJS.

ar2rsawseen replied:
In overall nodeenv is useless to Countly, since in production we have control over what is installed and nothing else should be running and preconfigured on the server. It will only be one more dependency to worry about when making offline builts with prebuild binaries, etc
But since here it is optional, should be ok, only not sure about the path, seems to be tailored to your system? Is there some default path to be used in such cases?

GlennAustin replied:
I'm running a production environment, with multiple versions of NodeJS required for other various low-volume services (most run in 12.x, but a few are running on 14.x). I suppose I could run either Count.ly or those other services in Docker containers, but that adds an overhead that is disproportionate to the load required.
The biggest problem is that Count.ly is currently now *several* versions behind on NodeJS - requiring 10.x while the most recent LTS is 12.x and release is 14.x.
